File: haskell/0002.hs
main = putStrLn $ show evenfibsum

evenfibsum = sum $ takeWhile (< 4000000) (filter even fibonacci)
    where
        -- shamelessly taken from the haskell wiki
        -- this sums an element with the next element in the sequence
        fibonacci = [0, 1] ++ zipWith (+) fibonacci (tail fibonacci)

File: python/0002.py
from itertools import takewhile
def evenfib():
    # Even :: F(n) = 4*F(n-1) + F(n-2); F(0)=0, F(1)=2
    i, j = 0, 2
    while 1:
        yield i
        i, j = j, 4*j + i

solution2 = sum(takewhile((4_000_000).__gt__, evenfib()))

print(solution2)